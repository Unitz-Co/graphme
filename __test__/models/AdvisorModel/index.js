import _ from 'lodash';
import graphme from '../../../index';

import Definition from '../Definition';

import GetAdvisor from '../../gql/advisor/GetAdvisor.gql';

import AdvisorProfileModel from '../AdvisorProfileModel';
import AdvisorPresenceModel from '../AdvisorPresenceModel';
import TransactionModel from '../TransactionModel';
import TransactionAggregateModel from '../TransactionAggregateModel';

class AdvisorModel extends graphme.BaseModel {
  static DEFINITION = Definition.create({
    name: 'Advisor',
    schema: {
      id: String,
      is_active: Boolean,
      created_at: Date,
    },
    nodes: [
      ['profile', 'AdvisorProfileModel', { column_mapping: { id: 'advisor_id' } }],
      ['presence', 'AdvisorPresenceModel', { column_mapping: { id: 'advisor_id' } }],
      ['transactions', 'TransactionModel.Collection', { column_mapping: { id: 'advisor_id' } }],
      [
        'transactions_aggregate',
        'TransactionAggregateModel',
        { column_mapping: { id: 'advisor_id' }, usePlanSync: true },
      ],
    ],
    key: 'id',

    baseQuery: GetAdvisor,

    GQL_ACTIONS: {
      FIND: `advisor`,
      GET: `advisor_by_pk`,
      INSERT: `insert_advisor_one`,
      UPDATE: `update_advisor_by_pk`,
      DELETE: `delete_advisor_by_pk`,
    },
  });
}

graphme.model('AdvisorModel', AdvisorModel);

export default AdvisorModel;
