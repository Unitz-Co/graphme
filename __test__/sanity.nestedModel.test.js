import _ from 'lodash';
import AdvisorModel from './models/AdvisorModel';

const instance = AdvisorModel.fromData({ id: 'PdOJWFBgNPUEMhX1JlsDm7zWy012' });

test('indirect nested model property getter', async () => {
  const profile = await instance.profile;
  console.snapshot('profile', profile);
  console.snapshot('profile.display_name (indirect)', await profile.display_name);
});

test('direct nested instance property getter', async () => {
  console.snapshot('profile.display_name (direct)', await instance.getByPath('profile.display_name'));
});

test('direct nested model property setter', async () => {
  const profile = await instance.profile;
  const oldName = await profile.display_name;

  console.snapshot('profile.display_name: (before)', oldName);
  const newName = `newVal_newName`;

  profile.display_name = newName;
  await profile.save();

  console.snapshot('profile.display_name: (change)', newName, await instance.getByPath('profile.display_name'));
  profile.display_name = oldName;
  await profile.save();

  console.snapshot('profile.display_name: (after)', await instance.getByPath('profile.display_name'));
});


test('direct nested model property setByPath', async () => {
  const oldName = await instance.getByPath('profile.display_name');
  console.snapshot('profile.display_name: (before)', oldName);
  const newName = `newVal_newName`;

  await instance.setByPath('profile.display_name', newName);
  await instance.applyByPath('profile.save');

  console.snapshot('profile.display_name: (change)', newName, await instance.getByPath('profile.display_name'));
  await instance.setByPath('profile.display_name', oldName);
  await instance.applyByPath('profile.save');

  console.snapshot('profile.display_name: (after)', await instance.getByPath('profile.display_name'));
});


test('direct nested model method call', async () => {
  const profile = await instance.profile;
  console.snapshot('getClass', profile.getClass());
});

test('direct nested model method applyByPath', async () => {
  console.snapshot('getClass', await instance.applyByPath('profile.getClass'));
});


