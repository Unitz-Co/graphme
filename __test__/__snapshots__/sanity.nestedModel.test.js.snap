// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direct nested instance property getter 1`] = `
Object {
  "args": Array [
    "profile.display_name (direct)",
    "new_displayName",
  ],
}
`;

exports[`direct nested model method applyByPath 1`] = `
Object {
  "args": Array [
    "getClass",
    [Function],
  ],
}
`;

exports[`direct nested model method call 1`] = `
Object {
  "args": Array [
    "getClass",
    [Function],
  ],
}
`;

exports[`direct nested model property setByPath 1`] = `
Object {
  "args": Array [
    "profile.display_name: (before)",
    "new_displayName",
  ],
}
`;

exports[`direct nested model property setByPath 2`] = `
Object {
  "args": Array [
    "profile.display_name: (change)",
    "newVal_newName",
    "newVal_newName",
  ],
}
`;

exports[`direct nested model property setByPath 3`] = `
Object {
  "args": Array [
    "profile.display_name: (after)",
    "new_displayName",
  ],
}
`;

exports[`direct nested model property setter 1`] = `
Object {
  "args": Array [
    "profile.display_name: (before)",
    "new_displayName",
  ],
}
`;

exports[`direct nested model property setter 2`] = `
Object {
  "args": Array [
    "profile.display_name: (change)",
    "newVal_newName",
    "newVal_newName",
  ],
}
`;

exports[`direct nested model property setter 3`] = `
Object {
  "args": Array [
    "profile.display_name: (after)",
    "new_displayName",
  ],
}
`;

exports[`indirect nested model property getter 1`] = `
Object {
  "args": Array [
    "profile",
    AdvisorProfileModel {},
  ],
}
`;

exports[`indirect nested model property getter 2`] = `
Object {
  "args": Array [
    "profile.display_name (indirect)",
    "new_displayName",
  ],
}
`;
